File: day01/py/b.py

from pathlib import Path

filename = '../input.txt'

def B(fname = filename):
    file = Path(fname)

    inc = 0
    ser = []

    with open(file) as f:
        for i in range(3):
            ser.insert(0,int(f.readline()))

        last = sum(ser)
        
        for l in f:
            i = int(l)

            ser.insert(0,i)
            ser.pop()
            if(sum(ser) > last):
                inc = inc + 1
            last = sum(ser)

    return inc

if __name__ == '__main__':
    print(B())

File: day01/cpp/b.cpp
/*
 *
 * 
 */

#include <iostream>
#include <fstream>
using namespace std;

int main()
{
	string fname = "../input.txt";
	int last = 0;
	int next = 0;
	int total = 0;
	bool first = true;

	ifstream input;
       
	input.open(fname);
	string temp = "";

	while(getline(input, temp))
	{
		next =  atoi(temp.c_str());
		if (first)
		{
			first = false;
		}
		else
		{
			if (next > last)
				total++;
		}
		last = next;

		
	}

	cout << total << '\n';
	return 0;
}
